### ISSUE TITLE
rollup-plugin-json isn't supported

### ISSUE THREAD

**Issue by TheBosZ** [PepsRyuu/nollup#17]
The error comes when I try to import the npm package `eddystone-url-encoding`:
```
(node:1988) UnhandledPromiseRejectionWarning: Error: "version" is a required argument.
 --- C:\projects\sample-project\node_modules\eddystone-url-encoding\lib\prefixes.json
 --- C:\projects\sample-project\node_modules\eddystone-url-encoding\lib\decode.js
 --- C:\projects\sample-project\node_modules\eddystone-url-encoding\index.js
 --- C:\projects\sample-project\src\utils\Beacon\Eddystone.ts
```

I'm **assuming** that the error is from the json plugin and not because the library is a CommonJS module, but, of course, I could be wrong.

**Comment by PepsRyuu** [comment]
Most likely related to the source map collapsing support I was working on yesterday. JSON plugin returns a source map object with empty mappings which I didn't check for. Should be an easy fix!

**Comment by PepsRyuu** [comment]
Should be fixed in the https://github.com/PepsRyuu/nollup/tree/Globals branch.

If you could just confirm from your end if it's resolved I'd appreciate it. :)

**Comment by PepsRyuu** [comment]
Not sure I can do anything about circular dependencies. Have you consider using ```rollup-plugin-alias``` and pointing to a built version of moment?

**Comment by PepsRyuu** [comment]
Released sourcemap fix in 0.3.0 which closes this issue. 

I've researched into circular dependencies, and it's not something I really want to support... The solution to circular dependencies is to implement ESM live bindings. Webpack accomplishes this by replacing all import variables with objects and uses Object defineProperty getters to get the latest value. It looks roughly like this:

```
// In the exported file, if we exported a variable called "number",
// it will add this to the exports instead of "number" directly.
Object.defineProperty(__webpack_exports__, "a", {
    configurable: false,
    enumerable: true,
    get: () => number
});

// Import variables are replaced to this renamed variable.
var __WEBPACK_IMPORTED_MODULE_0__a__ = __webpack_require__(0);

// All references to the imported variable must be replaced and does a get
// on the specially defined property. This property will return the latest value.
console.log(__WEBPACK_IMPORTED_MODULE_0__a__["a" /* number */]); // Calls
```

A couple of reasons I don't want to do this:

* Significant complexity, would have to traverse the entire syntax tree, figured out if the imported variable was used, and replace it with the objects. This can introduce a significant performance penalty, and might not be entirely reliable unless all possibilities have been considered (eg. what if a local variable by the same name as an import is declared?)

* Breaks debugging symbols. By swapping out import variables with objects, it breaks debugging because the variables you see in the code are not the variables that are outputted. So if you want to watch variables, or test stuff in the console, you have to use the transpiled variable name instead which can be confusing.

I recommend using ```rollup-plugin-alias``` and pointing to the compiled version of Moment, or using the new "globals" feature and importing Moment with a script tag. Using a built version of Moment would be preferable anyways to speed up compilation (there's a lot of files in moment's src!). 

For tracking purposes, I'll create a separate issue about circular dependencies (and repost what I said here), and will leave it open for someone to take if they want to. 

Thanks for bringing this issue to my attention! :)

**Comment by PepsRyuu** [comment]
https://github.com/PepsRyuu/nollup/issues/19 For tracking purposes! :)
